test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1992,6 +1998,9 @@ protected TestCluster buildTestCluster(Scope scope, long seed) throws IOExceptio
             }
             mockPlugins = mocks;
         }
+        if (addMockTelemetryPlugin()) {
```

Comment:
Please revert that, make a change in the 
```
protected Collection<Class<? extends Plugin>> getMockPlugins() {
```

Reply:
The idea here is we want to have MockTelemetryPlugin be included for all integ test cases by default, so to perform validations that none of the spans should have memory leak . 

getMockPlugins() can be overridden by specific integ test cases, we don't want that unless we want to include some other Telemetry plugin. That's why added addMockTelemetryPlugin() like other methods https://github.com/opensearch-project/OpenSearch/blob/e1a41255de203bc57c1e29469a9bb5d50fe0d84b/test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java#L2042 and override it in OTel plugin specific test case.